diff --git a/imgui.cpp b/imgui.cpp
index a1bde6e..6547f3a 100644
--- a/imgui.cpp
+++ b/imgui.cpp
@@ -16772,11 +16772,12 @@ static void ImGui::DockNodeUpdate(ImGuiDockNode* node)
     if (node->IsRootNode() && host_window)
     {
         DockNodeTreeUpdatePosSize(node, host_window->Pos, host_window->Size);
-        PushStyleColor(ImGuiCol_Separator, g.Style.Colors[ImGuiCol_Border]);
-        PushStyleColor(ImGuiCol_SeparatorActive, g.Style.Colors[ImGuiCol_ResizeGripActive]);
-        PushStyleColor(ImGuiCol_SeparatorHovered, g.Style.Colors[ImGuiCol_ResizeGripHovered]);
+        // custom change: disabled this
+        // PushStyleColor(ImGuiCol_Separator, g.Style.Colors[ImGuiCol_Border]);
+        // PushStyleColor(ImGuiCol_SeparatorActive, g.Style.Colors[ImGuiCol_ResizeGripActive]);
+        // PushStyleColor(ImGuiCol_SeparatorHovered, g.Style.Colors[ImGuiCol_ResizeGripHovered]);
         DockNodeTreeUpdateSplitter(node);
-        PopStyleColor(3);
+        // PopStyleColor(3);
     }
 
     // Draw empty node background (currently can only be the Central Node)
